proto/desmos/supply/v1/query.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+package desmos.supply.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/supply/types";
+
+
+// Query defines the gRPC querier service.
+service Query {
+
+  // TotalSupply queries the total supply of the given denom, converted with the given divider
+  rpc TotalSupply(QueryTotalSupplyRequest)
+      returns (QueryTotalSupplyResponse) {
+    option (google.api.http).get = "/desmos/supply/v1/total-supply/{denom}/{divider}";
```

Comment:
Since the `divider` is optional, it's much better to leave it as a simple [query parameter](https://docs.oracle.com/en/cloud/saas/cx-commerce/21d/ccdev/rest-api-query-parameters.html). To do this, simply remove it from the URL: 
```suggestion
    option (google.api.http).get = "/desmos/supply/v1/total-supply/{denom}";
```

The call will now need to be 
```
/desmos/supply/v1/total-supply/dsm?divider={divider}
```
instead of 
```
/desmos/supply/v1/total-supply/dsm/10000
```